libraries/shared/mocks/storage_queue.go

```diff
@@ -20,27 +20,40 @@ import (
 	"github.com/vulcanize/vulcanizedb/libraries/shared/storage/utils"
 )
 
+// MockStorageQueue for tests
 type MockStorageQueue struct {
-	AddCalled      bool
-	AddError       error
-	AddPassedDiff  utils.StorageDiff
-	DeleteErr      error
-	DeletePassedId int
-	GetAllErr      error
-	DiffsToReturn  []utils.StorageDiff
+	AddCalled       bool
+	AddError        error
+	AddPassedDiffs  []utils.StorageDiff
+	DeleteErr       error
+	DeletePassedIds []int
+	GetAllErr       error
+	DiffsToReturn   []utils.StorageDiff
+	GetAllCalled    bool
 }
 
+// Add mock method
 func (queue *MockStorageQueue) Add(diff utils.StorageDiff) error {
 	queue.AddCalled = true
-	queue.AddPassedDiff = diff
+	queue.AddPassedDiffs = append(queue.AddPassedDiffs, diff)
 	return queue.AddError
 }
 
+// Delete mock method
 func (queue *MockStorageQueue) Delete(id int) error {
-	queue.DeletePassedId = id
+	queue.DeletePassedIds = append(queue.DeletePassedIds, id)
+	diffsToReturn := make([]utils.StorageDiff, 0)
+	for _, diff := range queue.DiffsToReturn {
+		if diff.Id != id {
+			diffsToReturn = append(diffsToReturn, diff)
+		}
```

Comment:
This implementation makes me think that should be a map instead of a slice.

Reply:
Sure, can do, didn't want to mess with the other tests that depend on this mock too much.